main/src/ca/uwaterloo/flix/language/ast/NamedAst.scala

```diff
@@ -255,6 +257,20 @@ object NamedAst {
 
   }
 
+  sealed trait RestrictableChoicePattern
+
+  object RestrictableChoicePattern {
+
+    sealed trait VarPlace
+
+    case class Wild(loc: SourceLocation) extends VarPlace
+
+    case class Var(sym: Symbol.VarSym, loc: SourceLocation) extends VarPlace
+
+    case class Tag(qname: Name.QName, pat: List[VarPlace], loc: SourceLocation) extends RestrictableChoicePattern
```

Comment:
Looks like `RestrictableChoicePattern` is an enum with only one member? Can it then just be a case class?

Reply:
It could yes but I felt that this was more extensible and follows the other patterns